tests/test_document.py

```diff
@@ -230,6 +230,36 @@ class CustomIdInheritance(CustomId):
 
         assert 'id' not in CustomIdInheritance.schema.fields
 
+    def test_is_modified(self):
+
+        @self.instance.register
+        class Vehicle(EmbeddedDocument):
+            name = fields.StrField()
+
+        @self.instance.register
+        class Driver(Document):
+            name = fields.StrField()
+            vehicle = fields.EmbeddedField(Vehicle)
+
+        driver = Driver(name='Marty')
```

Comment:
This document has not been commited yet, thus it should be `is_modified() == True`

You should check `is_created` within `is_modified` for this.